File: backend/internal/model/stats.go
package model

type UserStats struct {
	CardsLogged  int      `json:"cards_logged"`
	BestScore    *int16   `json:"best_score,omitempty"`
	BestXCount   *int16   `json:"best_x_count,omitempty"`
	AvgScore     *float64 `json:"avg_score,omitempty"`
	Rolling10Avg *float64 `json:"rolling_10_avg,omitempty"`
}

type RifleStats struct {
	RifleID    string `json:"rifle_id"`
	BestScore  int16  `json:"best_score"`
	BestXCount int16  `json:"best_x_count"`
	CardCount  int    `json:"card_count"`
}

type ScoreTrendPoint struct {
	Period     string  `json:"period"`
	AvgScore   float64 `json:"avg_score"`
	BestScore  *int16  `json:"best_score,omitempty"`
	StdDev     float64 `json:"std_dev"`
	CardCount  int     `json:"card_count"`
	AvgXCount  float64 `json:"avg_x_count"`
	BestXCount *int16  `json:"best_x_count,omitempty"`
}

type ComboPerformanceSummary struct {
	RifleID     string   `json:"rifle_id"`
	RifleName   string   `json:"rifle_name"`
	PelletID    string   `json:"pellet_id"`
	PelletName  string   `json:"pellet_name"`
	BestGroupMM *float64 `json:"best_group_mm,omitempty"`
	AvgGroupMM  *float64 `json:"avg_group_mm,omitempty"`
	TestCount   int      `json:"test_count"`
	Consistency *float64 `json:"consistency,omitempty"`
	LastTested  string   `json:"last_tested"`
}

File: frontend/src/api/stats.ts
import { api } from './client'

export interface UserStats {
  cards_logged: number
  best_score?: number
  best_x_count?: number
  avg_score?: number
  rolling_10_avg?: number
}

export interface RifleStats {
  rifle_id: string
  best_score: number
  best_x_count: number
  card_count: number
}

export interface ScoreTrendPoint {
  period: string
  avg_score: number
  best_score?: number
  std_dev: number
  card_count: number
  avg_x_count: number
  best_x_count?: number
}

export const statsApi = {
  getMe: () => api.get<UserStats>('/users/me/stats'),
  getRifleStats: () => api.get<{ items: RifleStats[] }>('/users/me/rifle-stats'),
  getScoreTrends: (period: 'week' | 'month' = 'week', rifleId?: string) => {
    const params = new URLSearchParams({ period })
    if (rifleId) params.set('rifle_id', rifleId)
    return api.get<{ items: ScoreTrendPoint[] }>(`/users/me/score-trends?${params}`)
  },
}
